ipsource.go

```diff
@@ -313,41 +314,62 @@ func (c Command) GetIPs(ctx context.Context, versions IPVersions) ([]net.IP, err
 		expandedArgs[i] = repl.ReplaceAll(c.Args[i], "")
 	}
 
-	var cancel context.CancelFunc
 	if c.Timeout > 0 {
 		ctx, cancel = context.WithTimeout(ctx, time.Duration(c.Timeout))
 	}
 
 	cmd := exec.CommandContext(ctx, c.Cmd, expandedArgs...)
 	cmd.Dir = c.Dir
-
 	cmd.Stdout = &stdout
 	cmd.Stderr = &stderr
 
 	if cancel != nil {
 		defer cancel()
 	}
 
+	c.logger.Debug("running command",
+		zap.String("command", c.Cmd),
+		zap.Strings("args", expandedArgs),
+		zap.String("dir", c.Dir),
+		zap.Int64("timeout", int64(time.Duration(c.Timeout))),
+	)
+
 	err := cmd.Run()
+	if err != nil {
+		return nil, err
+	}
 
 	exitCode := cmd.ProcessState.ExitCode()
-	if exitCode != 0 {
-		return nil, fmt.Errorf("Process exit with: %d", exitCode)
+	if exitCode != 0 || len(stderr.String()) > 0 {
+		c.logger.Error("command execution failed",
+			zap.String("command", c.Cmd),
+			zap.Strings("args", expandedArgs),
+			zap.String("dir", c.Dir),
+			zap.String("stdout", stdout.String()),
+			zap.String("stderr", stderr.String()),
+			zap.Int("exit code", exitCode))
+		return nil, fmt.Errorf("command %s exited with: %d", c.Cmd, exitCode)
 	}
 
-	if err != nil {
-		return nil, err
-	}
-	// Here we need to evalutate StdOut and StdErr
 	ipStr := strings.TrimSpace(string(stdout.String()))
 	ipArr := strings.Split(ipStr, ",")
```

Comment:
Actually I think we should split _then_ trim (loop over the values), cause then that would allow `1.1.1.1, 2.2.2.2`, i.e. spaces before/after commas)